### ISSUE TITLE
Should the content-type header be allowed to end with a semi-colon (;)

### ISSUE THREAD

**Issue by Ilyes512** [neomerx/json-api#224]
Assuming that `$headerParametersParser` is an instance of `\Neomerx\JsonApi\Http\Headers\HeaderParametersParser` I get an error when parsing a conten-type header that ends with a `;` (semi-colon). I do no not get that when parsing an accept header that end's with a `;`. In both cases the header's don't actually have any media type parameters.

```php
// Parsing content-type header:
$headerParametersParser->parseContentTypeHeader('application/vnd.api+json'); // this is fine
// but:
$headerParametersParser->parseContentTypeHeader('application/vnd.api+json;'); // this will return a InvalidArgumentException

// Parsing accept header:
$headerParametersParser->parseAcceptHeader('application/vnd.api+json'); // this fine
$headerParametersParser->parseAcceptHeader('application/vnd.api+json;'); // this is also fine
```

So not sure what the spec says exactly about [content-type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) and [accept](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept). (The original RFC's are linked at the bottom).

Not sure how to read the grammer used to describe the "syntax", but (thinking logically) I think they both should respond the same (so either both should throw an InvalidArgumentException or not).

**Comment by neomerx** [comment]
The difference in behavior was just implementation specifics. I've added a test for the new value and fixed the code.

**Comment by neomerx** [comment]
released as `v3.0.0-rc.3`

**Comment by neomerx** [comment]
released as [v3.0.0](https://github.com/neomerx/json-api/releases/tag/v3.0.0)
